Source/Core/Common/Logging/LogManager.cpp

```diff
@@ -123,10 +124,19 @@ void LogManager::Log(LogTypes::LOG_LEVELS level, LogTypes::LOG_TYPE type, const
   if (!log->IsEnabled() || level > log->GetLevel() || !log->HasListeners())
     return;
 
+  static size_t pos;
+  std::string pattern = DIR_SEP "Source" DIR_SEP "Core" DIR_SEP;
+  if (!pos)
+  {
+    pos = static_cast<std::string>(file).find(pattern);
```

Comment:
this can just be `std::string(file)`, or you can alter `Log` to pass the file in as a std::string.